docs/src/main/asciidoc/building-my-first-extension-guide.adoc

```diff
@@ -0,0 +1,700 @@
+At this point, you just need to understand that this class explains to Quarkus how to deploy a feature named `greeting` which is your extension.
+In other words, you are augmenting your application to use the `greeting` extension with all the Quarkus benefits (build time optimization, native support, etc.).
+
+=== The Runtime module
+
+Finally `./greeting/runtime/pom.xml`.
+
+[source, xml]
+-------------
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.acme</groupId>
+        <artifactId>quarkus-greeting-parent</artifactId>
+        <version>1.0-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>quarkus-greeting</artifactId>  <!--1-->
+    <name>Greeting Extension - Runtime</name>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>io.quarkus</groupId>
+                <artifactId>quarkus-bootstrap-maven-plugin</artifactId>  <!--2-->
+                <version>${quarkus.version}</version>
+                <executions>
+                    <execution>
+                        <goals>
+                            <goal>extension-descriptor</goal>
+                        </goals>
+                        <phase>compile</phase>
+                        <configuration>
+                            <deployment>${project.groupId}:${project.artifactId}-deployment:${project.version}
+                            </deployment>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <configuration>
+                    <annotationProcessorPaths>
+                        <path>
+                            <groupId>io.quarkus</groupId>
+                            <artifactId>quarkus-extension-processor</artifactId> <!--3-->
+                            <version>${quarkus.version}</version>
+                        </path>
+                    </annotationProcessorPaths>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+</project>
+-------------
+
+The key points are:
+
+<1> By convention, the runtime module has no suffix (`greeting`) as it is the artifact exposed to the end user.
+<2> We add  the `quarkus-bootstrap-maven-plugin` to generate the Quarkus extension descriptor included into the runtime artifact which links it with the corresponding deployment artifact.
+<3> We add  the `quarkus-extension-processor` to the compiler annotation processors.
+
```

Comment:
Nope. it's attached to the runtime module compilation
```
            <plugin>
                <groupId>io.quarkus</groupId>
                <artifactId>quarkus-bootstrap-maven-plugin</artifactId>
                <version>${quarkus.version}</version>
                <executions>
                    <execution>
                        <goals>
                            <goal>extension-descriptor</goal>
                        </goals>
                        <phase>compile</phase>
                        <configuration>
                            <deployment>${project.groupId}:${project.artifactId}-deployment:${project.version}
                            </deployment>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
```
`extension-descriptor` it generates a `quarkus-extension.yaml` file:
```
---
metadata: {}
group-id: "org.acme"
artifact-id: "quarkus-greeting"
version: "1.0-SNAPSHOT"
name: "Greeting Extension - Runtime"
```
To be honest, I don't even know how it can be configured via Maven properties

Reply:
Based on https://quarkus.io/guides/writing-extensions
> There is one step (specific to the Quarkus source tree) that you should do manually when creating a new extension: create a quarkus-extension.yaml file that describe your extension inside the runtime module src/main/resources/META-INF folder.

Assuming that the most of the future extensions will be hosted outside Quarkus repo, can we skip this point in this guide? IMHO, this action will be part of a Quarkus universe integration.
WDYT?